roteiro-de-estudos/algorithm-analysis.md

```diff
@@ -0,0 +1,414 @@
+
+<div style="text-align:center">
+  <img src="https://render.githubusercontent.com/render/math?math=a_{k}%20=%20a_{1}%20\cdot%20q^{k-1};%20">
+</div>
+
+<div style="text-align:center">
+  <img src="https://render.githubusercontent.com/render/math?math=a_{1}%20=%20n%20%3B%20%20q%20=%20\dfrac{1}{2}%20%3B%20k%20=%20i%20%3B">
+</div>
+
+<div style="text-align:center">
+  <img src="https://render.githubusercontent.com/render/math?math=1%20=%20n%20\cdot%20(\dfrac{1}{2})^{i-1}">
+</div>
+
+<div style="text-align:center">
+  <img src="https://render.githubusercontent.com/render/math?math=2^{i-1}%20=%20n">
+</div>
+
+Aplicando logaritmo na base 2 em ambos os lados da equação, temos o número de repetições **i** em função do tamanho do valor **n**
+
+<div style="text-align:center">
+  <img src="https://render.githubusercontent.com/render/math?math=\log_{2}{(n)}%20=%20i%20-%201">
+</div>
+
+<div style="text-align:center">
+  <img src="https://render.githubusercontent.com/render/math?math=\log_{2}{(n)} + 1 = i ">
+</div>
+
+Por fim, dado que as operações 2 e 4 estão dentro de um loop que se repete **i** vezes, e sendo **i** representado através do valor da entrada, é possível escrever a função **Op<sub>Total</sub>(n)** para o algoritmo abaixo como sendo
+
+<div style="text-align:center">
+  <img src="https://render.githubusercontent.com/render/math?math=\textrm{Op}_{\textrm{total}}(n)%20=%202%20%2B%20\log{2}{(n)}%20%2B%201%20%2B%20\log{2}{(n)}%20%2B%201%20=%20%20%204%20%2B%202%20\cdot%20\log{2}{(n)}">
+</div>
+
+Após solucionar o problema 2, percebe-se que nem sempre o número de execuções de uma operação é igual ao tamanho do parâmetro de entrada do algoritmo, dependendo da maneira como as condições do código são impostas, a quantidade de operações executadas pode ser escrita em função da variável de entrada de outras maneiras. Além disso, ao comparar a função **Op<sub>Total</sub>(n)** do problema atual com as funções calculadas anteriormente, percebemos que este algoritmo executa menos operações que todos os outros algoritmos mostrados acima para valores elevados de **n**, para **n = 32 o algoritmo do problema atual executa 14 operações ao total, enquanto que o algoritmo do problema 1 executa 1192 operações, por exemplo**.
+
+A seguir, analisaremos um último algoritmo a fim de identificar um comportamento interessante neste.
+
+# Problema 3
```

Comment:
Se for fazer uma função recursiva, sugiro deixar isso pra um outro artigo. Senão a leitura vai ficar muito grande, e pra uma introdução à análise de algoritmos o ponto chave são os laços de repetição.